library/ccm.c

```diff
@@ -192,23 +246,28 @@ static int ccm_auth_crypt( mbedtls_ccm_context *ctx, int mode, size_t length,
      * 5 .. 3   (t - 2) / 2
      * 2 .. 0   q - 1
      */
-    b[0] = 0;
```

Comment:
I hoped that `ccm_init()` will take care of clearing the `b[0]`, but it turns out that this has to be cleared inside either
`ccm_starts()` or `ccm_set_lengths()`. A self-test function at the bottom of this file is calling `ccm_init()` only once, and then performing encrypt/decrypt cycle on some test data. This means that the `b[0]` is not cleared between `mbedtls_ccm_encrypt_and_tag` and `mbedtls_ccm_auth_decrypt`

This can be tackled in two ways:
1. Move b[0] calculation to `mbedtls_ccm_calculate_first_block`. This will require storing all necessary variables in the ccm_context
2. Create additional auxiliary function, to conditionally clear the b[0] at the beginning of `ccm_starts` and `ccm_set_lengths` (same mechanism as in line 151)

Reply:
Update - went with second approach